src/engraving/libmscore/instrument.cpp

```diff
@@ -988,6 +940,10 @@ void InstrChannel::read(XmlReader& e, Part* part)
     }
 
     _mustUpdateInit = true;
+    if (e.context()
+        && e.context()->audioSettingCompat().count(partAudioSetting.instrumentId.partId) == 0) {
+        e.context()->addPartAudioSettingCompat(partAudioSetting);
```

Comment:
I think it's better to hide this logic with a check that the data is already there in the `addPartAudioSettingCompat` method, we can call it `addUniquePartAudioSettingCompat`

Reply:
fixed